fhirstore/search/search_methods.py

```diff
@@ -54,16 +59,33 @@ def build_element_query(key, value):
         elif string_modifier == "below":
             element_query["simple_query_string"]["query"] = f"({value})*"
             element_query["simple_query_string"]["fields"] = [string_field]
+        elif string_modifier == "identifier":
+            element_query["simple_query_string"]["query"] = value
+            element_query["simple_query_string"]["fields"] = [
+                f"{string_field}.identifier.value"
+            ]
 
-    elif numeric_modif:
+    elif numeric_prefix:
         element_query["range"][key] = {
-            number_prefix_matching[numeric_modif.group(1)]: numeric_modif.group(2)
+            number_prefix_matching[numeric_prefix.group(1)]: numeric_prefix.group(2)
         }
-    elif eq_modif:
-        element_query["match"][key] = eq_modif.group(2)
-    else:
+    elif eq_prefix or isinstance(value, int) or isinstance(value, float):
```

Comment:
Could you explain a bit what happens here?
Also, the case `isinstance(value, int)` is handled in the last `elif`

Reply:
When the value is an int, a float or if there is the modifier "eq", the corresponding elasticsearch query is {"match": {"key": "value"}} . I was trying to group the three cases here, but I don't think its the best way to do it.

Reply:
I find it clearer now that you've separated these cases :)